Задача по подсчету островов (islands.cpp)

SlavaSTiwari opened the pull request:
Задача нахождения числа островов в матрице. Я решил его с помощью "Depth-First Search (DFS)" и рекурсии.

SlavaSTiwari committed:
Задача по подсчету островов (islands.cpp)

Задача нахождения числа островов в матрице. Я решил его с помощью "Depth-First Search (DFS)" и рекурсии.

SorokinIA commented on islands.cpp:
```diff
@@ -0,0 +1,76 @@
+void DFS(int x, int y, Vector_2d &world_map, Vector_2d_bool &visited) {
+
+    // arrays to make check connected land tiles:
+    int y_move[] = {-1, -1, -1, 0, 0, 1, 1, 1};
```
Нет нет в условии только вверх вниз вправо влево , углы не надо

SorokinIA commented on islands.cpp:
```diff
@@ -0,0 +1,76 @@
+
+    // arrays to make check connected land tiles:
+    int y_move[] = {-1, -1, -1, 0, 0, 1, 1, 1};
+    int x_move[] = {-1, 0, 1, -1, 1, -1, 0, 1};
```
Вот зачем тебе обычный array есть же vector с ним и работать удобнее, но если совсем хочешь по памяти запарится есть array как структура по крайней мере числа не придётся хардкодить

SorokinIA commented on islands.cpp:
```diff
@@ -0,0 +1,76 @@
+        {1,1,0,0,0}  // 2
+    };
+
+    Vector_2d_bool visited(w_map.size(), std::vector<bool>(w_map[0].size(), 0));
```
Можно и в обычный было просто передавать точки которые относятся только к одному острову...

SlavaSTiwari committed:
Островной счетчик (ИСПРАВЛЕНО) (islands_2.cpp)

Удалены ненужные массивы.
Теперь ищет только острова в 4 направлениях (не углы).
Удалена вторая "visited" матрица, теперь используется только исходная карта для привязки и подсчета.
